src/main/java/edu/iastate/ece/sd/sdmay2126/runner/selenium/SeleniumUtilities.java

```diff
@@ -0,0 +1,93 @@
+package edu.iastate.ece.sd.sdmay2126.runner.selenium;
+
+import org.openqa.selenium.*;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+import java.util.List;
+
+/**
+ * Provides basic, composable Selenium operations.
+ */
+public class SeleniumUtilities {
+    /** Reduce spinning while waiting. */
+    private static final int POLLING_DELAY_MILLIS = 50;
+
+    /**
+     * Waits for an element's visibility to change.
+     * @param element The element to watch.
+     * @param desiredVisibility The visibility to wait for.
+     * @param maxWait How long to wait for.
+     */
+    public static void waitForVisibilityChange(WebElement element, boolean desiredVisibility, Duration maxWait)
+            throws SeleniumIdentificationException {
+        long startTime = System.currentTimeMillis();
+        try {
+            // Loop while element doesn't match desired visibility until max time elapsed
+            while (desiredVisibility != element.isDisplayed()) {
+                // Fail if we've exceeded our wait period
+                if (System.currentTimeMillis() - startTime >= maxWait.toMillis()) {
+                    throw new SeleniumIdentificationException("Failed to detect a visibility change within the wait period.");
+                }
+
+                // Reduce spinning with polling delay
+                Thread.sleep(POLLING_DELAY_MILLIS);
+            }
+        } catch (StaleElementReferenceException | InterruptedException e) {
+            // In cases where desiredVisibility is false, a StaleElementReferenceException may be thrown
+        }
+    }
+
+    /**
+     * Will attempt to click something, even if obscured, until successful. Note that this is necessary since we can't
+     * determine if something is obscured until we try to interact with it.
+     * @param element The element to click.
+     * @param maxWait How long to try before failing.
+     */
+    public static void clickUntilSuccessful(WebElement element, Duration maxWait) throws InterruptedException,
```

Comment:
This is interesting. I didn't realize that selenium would throw those exceptions on non-clickable elements.

Reply:
It appears that Selenium compares an expected result (button `onclick` is triggered) against the reality at runtime. In this case, we were seeing errors from the button being obscured and the click event not being triggered.

It was even smart-enough to identify that the button was obscured. Just a shame that it can't do that check _before_ we try to click. :)